x-pack/plugins/security_solution/common/types/timeline/api.ts

```diff
@@ -320,11 +320,11 @@ export const TimelineSavedToReturnObjectRuntimeType = runtimeTypes.intersection(
     version: runtimeTypes.string,
   }),
   runtimeTypes.partial({
-    eventIdToNoteIds: runtimeTypes.array(NoteSavedObjectToReturnRuntimeType),
+    eventIdToNoteIds: runtimeTypes.array(NoteRuntimeType),
     noteIds: runtimeTypes.array(runtimeTypes.string),
-    notes: runtimeTypes.array(NoteSavedObjectToReturnRuntimeType),
+    notes: runtimeTypes.array(NoteRuntimeType),
     pinnedEventIds: runtimeTypes.array(runtimeTypes.string),
-    pinnedEventsSaveObject: runtimeTypes.array(PinnedEventToReturnSavedObjectRuntimeType),
+    pinnedEventsSaveObject: runtimeTypes.array(PinnedEventRuntimeType),
```

Comment:
Sorry I am commenting on something you have not touched. But since we are refactoring, I thought of bringing it up.

As mentioned in one of my comment below.. the type `TimelineSavedObject` 4 lines below this code seems to be somewhat misleading as it is actually a `TimelineSavedToReturnObjectRuntimeType`. Actual saved object does not include `notes`  or `pinnedEvents` related data.